gplugins/lumerical/fdtd.py

```diff
@@ -0,0 +1,1282 @@
+                background_doping_concentration=None,
+                background_doping_ion=None,
+                orientation="100",
+            ),
+            "box": LayerLevel(
+                name=None,
+                layer=(99999, 0),
+                thickness=3.0,
+                thickness_tolerance=None,
+                zmin=-3.0,
+                zmin_tolerance=None,
+                material="sio2",
+                sidewall_angle=0.0,
+                sidewall_angle_tolerance=None,
+                width_to_z=0.0,
+                z_to_bias=None,
+                mesh_order=9,
+                layer_type="background",
+                mode=None,
+                into=None,
+                resistivity=None,
+                bias=None,
+                derived_layer=None,
+                info={},
+                background_doping_concentration=None,
+                background_doping_ion=None,
+                orientation="100",
+            ),
+            "core": LayerLevel(
+                name=None,
+                layer=(1, 0),
+                thickness=0.22,
+                thickness_tolerance=None,
+                zmin=0.0,
+                zmin_tolerance=None,
+                material="si",
+                sidewall_angle=10.0,
+                sidewall_angle_tolerance=None,
+                width_to_z=0.5,
+                z_to_bias=None,
+                mesh_order=2,
+                layer_type="grow",
+                mode=None,
+                into=None,
+                resistivity=None,
+                bias=None,
+                derived_layer=None,
+                info={"active": True},
+                background_doping_concentration=100000000000000.0,
+                background_doping_ion="Boron",
+                orientation="100",
+            ),
+            # KNOWN ISSUE: Lumerical 2021 version of Layer Builder does not support dopants in process file
+        }
+    )
+    SIMULATION_SETTINGS_LUMERICAL_FDTD.port_translation = 1.0
+    sim = LumericalFdtdSimulation(
+        component=taper,
+        layerstack=layerstack_lumerical2021,
+        convergence_settings=LUMERICAL_FDTD_CONVERGENCE_SETTINGS,
+        simulation_settings=SIMULATION_SETTINGS_LUMERICAL_FDTD,
+        hide=False,
+        run_port_convergence=False,
+        run_mesh_convergence=False,
+    )
+
+    sp = sim.write_sparameters(overwrite=True)
+    print(sp)
+    print("Done")
+
+
+class LumericalFdtdSimulation:
+    """
+    Lumerical FDTD simulation
+
+    Set up FDTD simulation based on component geometry and simulation settings. Optionally, run convergence.
+
+    Attributes:
+        component: Component geometry to simulate
+        layerstack: PDK layerstack
+        session: Lumerical session
+        simulation_settings: FDTD simulation settings
+        convergence_settings: FDTD convergence settings
+        dirpath: Directory where simulation files are saved
+        filepath_npz: S-parameter filepath (npz)
```

Comment:
these are output filepaths, right? should specify

Reply:
Done in latest:
https://github.com/seanlam97/gplugins/blob/a135b4bfcfbe6db972d8e3297c5291c593369a0d/gplugins/lumerical/fdtd.py